vertx-web/src/test/java/io/vertx/ext/web/handler/sockjs/SockJSHandlerTest.java

```diff
@@ -405,9 +404,7 @@ public void testWebContext() throws Exception {
         sock.routingContext().setSession(session);
         try {
           assertEquals(sessionID.get(), store.get(sessionID.get()).result().id());
-          // the user cannot be the same, it's a new request, if it's not null it
-          // has been leaked from another context
-          assertNull(sock.webUser());
+          assertEquals(sessionUser.get(), sock.webUser());
```

Comment:
The timing for the test will be correct if we don't discard the callback/future.